src/constructors/regular.c

```diff
@@ -134,6 +134,145 @@ igraph_error_t igraph_star(igraph_t *graph, igraph_integer_t n, igraph_star_mode
 }
 
+/**
+ * \ingroup generators
+ * \function igraph_wheel
+ * \brief Creates a \em wheel graph, a union of a star graph and a
+ *        cycle graph. Note that in case of a graph with 2 vertices,
+ *        the non-center vertex will have a self loop (may imagine
+ *        the earth is the center of the wheel, the moon and its orbit
+ *        is the rim of the wheel).
+ * \param graph Pointer to an uninitialized graph object, this will
+ *        be the result.
+ * \param n Integer constant, the number of vertices in the graph.
+ * \param mode Constant, gives the type of the star graph to
+ *        create. Possible values:
+ *        \clist
+ *        \cli IGRAPH_WHEEL_OUT
+ *          directed wheel graph, edges point
+ *          \em from the center to the other vertices.
+ *        \cli IGRAPH_WHEEL_IN
+ *          directed wheel graph, edges point
+ *          \em to the center from the other vertices.
+ *        \cli IGRAPH_WHEEL_MUTUAL
+ *          directed wheel graph with mutual edges.
+ *        \cli IGRAPH_WHEEL_UNDIRECTED
+ *          an undirected wheel graph is
+ *          created.
+ *        \endclist
+ * \param center Id of the vertex which will be the center of the
+ *          graph.
+ * \return Error code:
+ *         \clist
+ *         \cli IGRAPH_EINVVID
+ *           invalid number of vertices.
+ *         \cli IGRAPH_EINVAL
+ *           invalid center vertex.
+ *         \cli IGRAPH_EINVMODE
+ *           invalid mode argument.
+ *         \endclist
+ *
+ * Time complexity: O(|V|), the
+ * number of vertices in the graph.
+ *
+ * \sa \ref igraph_lattice(), \ref igraph_ring(), \ref igraph_star, \ref igraph_tree()
+ * for creating other regular structures.
+ *
+ */
+
+igraph_error_t igraph_wheel(igraph_t *graph, igraph_integer_t n, igraph_wheel_mode_t mode,
+                igraph_integer_t center) {
+
+    igraph_integer_t  v;
+    igraph_star_mode_t star_mode;
+    igraph_vector_int_t rim_edges = IGRAPH_VECTOR_NULL;
+    igraph_integer_t i;
+
+    /* create a star and also make use of its existing pre-check */
+    /* works for all input parameters                            */
+
+    switch(mode)
+    {
+        case IGRAPH_WHEEL_OUT:
+            star_mode = IGRAPH_STAR_OUT;
+            break;
+        case IGRAPH_WHEEL_IN:
+            star_mode = IGRAPH_STAR_IN;
+            break;
+        case IGRAPH_WHEEL_MUTUAL:
+            star_mode = IGRAPH_STAR_MUTUAL;
+            break;
+        case IGRAPH_WHEEL_UNDIRECTED:
+            star_mode = IGRAPH_STAR_UNDIRECTED;
+            break;
+        default:
+            IGRAPH_ERROR("invalid mode", IGRAPH_EINVMODE);
+    }
+
+    IGRAPH_CHECK(igraph_star(graph, n, star_mode, center));
+
+    /* If n <= 1, wheel graph is identical with star graph, */
+    /* no further processing is needed                      */
+
+    if (n <= 1) {
+        return IGRAPH_SUCCESS;
+    }
+
+    /* Add edges to the rim */
+
+    IGRAPH_VECTOR_INT_INIT_FINALLY(&rim_edges, 2 * (n-1));
```

Comment:
Please add a comment here and note that since we are adding at most as many edges as the star graph already has, `2*(n-1)` cannot overflow.

@ntamas @GroteGnoom Just a reminder that now we have overflow-safe integer operations, and almost all graph constructor should use them when creating the `edges` vector. _Usually_ calculations like `2*n` are at risk of overflow.

Reply:
Comments are added. However, as I don't quite understand the overflow operation, only explanation of 2 * (n-1) is added. If explanation about handling of overflow is needed, grateful if some guidance would be available.